src/runtime_src/core/tools/common/BusyBar.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+#include "BusyBar.h"
+
+// Local - Include Files
+#include "EscapeCodes.h"
+
+// 3rd Party Library - Include Files
+#include <boost/format.hpp>
+
+// System - Include Files
+#include <iostream>
+
+// ------ N A M E S P A C E ---------------------------------------------------
+using namespace XBUtilities;
+
+// ------ S T A T I C   V A R I A B L E S -------------------------------------
+static unsigned int busy_bar_width = 20;
+
+static boost::format fmtUpdate(EscapeCodes::cursor().hide()+
+                               EscapeCodes::fgcolor::reset()+ "[" +
+                               // This formats the width of progress bar and accepts a string
+                               // It breaks down into this %-20s which accepts a string that is 20 characters long
+                               EscapeCodes::fgcolor(EscapeCodes::FGC_IN_PROGRESS).string() + "%-" + std::to_string(busy_bar_width)+ "s" +
+                               EscapeCodes::fgcolor::reset()+ "]" +
+                               EscapeCodes::fgcolor(EscapeCodes::FGC_IN_PROGRESS).string() +
+                               EscapeCodes::fgcolor::reset()+ ": %s... < %s >\n"
+                              );
+
+BusyBar::BusyBar(const std::string &op_name, std::ostream &output)
+  : m_op_name(op_name)
+  , m_iteration(0)
+  , m_is_running(false)
+  , m_output(output)
+{}
+
+BusyBar::~BusyBar()
+{
+  // Make sure the thread has been collected and the bar text is removed
+  finish();
+}
+
+bool
+BusyBar::start(const bool is_batch)
+{
+  std::lock_guard lock(m_data_guard);
+  if (m_is_running)
+    return false;
```

Comment:
```suggestion
    return false;
```
Make this an exception